app/controllers/api/v3/bikes.rb

```diff
@@ -0,0 +1,176 @@
+          error!("You do not own that #{@bike.type}#{addendum}", 403)
+        end
+
+        def ensure_required_stolen_attrs(hash)
+          return true unless hash['bike']['stolen']
+          %w(phone city).each do |k|
+            error!("Could not create stolen record: missing #{k}", 401) unless hash['stolen_record'][k].present?
+          end
+        end
+      end
+
+      resource :bikes do
+        desc "View bike with a given ID"
+        params do
+          requires :id, type: Integer, desc: 'Bike id'
+        end
+        get ':id', serializer: BikeV2ShowSerializer, root:  'bike' do 
+          find_bike
+        end
+
+
+        desc "Add a bike to the Index!<span class='accstr'>*</span>", {
+          authorizations: { oauth2: [{ scope: :write_bikes }] },
+          notes: <<-NOTE
+            **Requires** `write_bikes` **in the access token** you use to create the bike.
+
+            <hr> 
+
+            **Creating test bikes**: To create test bikes, set `test` to true. These bikes:
+
+            - Do not show turn up in searches
+            - Do not send an email to the owner on creation
+            - Are automatically deleted after a few days
+            - Can be viewed in the API /v2/bikes/{id} (same as non-test bikes)
+            - Can be viewed on the HTML site /bikes/{id} (same as non-test bikes)
+
+            *`test` is automatically marked true on this documentation page. Set it to false it if you want to create actual bikes*
+
+            **Ownership**: Bikes you create will be created by the user token you authenticate with, but they will be sent to the email address you specify.
+
+          NOTE
+        }
+        params do
+          requires :serial, type: String, desc: "The serial number for the bike"
+          requires :manufacturer, type: String, desc: "Manufacturer name or ID"
+          # [Manufacturer name or ID](api_v2#!/manufacturers/GET_version_manufacturers_format)
+          requires :owner_email, type: String, desc: "Owner email"
+          requires :color, type: String, desc: "Main color or paint - does not have to be one of the accepted colors"
+          optional :test, type: Boolean, desc: "Is this a test bike?"
+          optional :organization_slug, type: String, desc: "Organization bike should be created by. **Only works** if user is a member of the organization"
+          optional :cycle_type_name, type: String, values: CYCLE_TYPE_NAMES, default: 'bike', desc: "Type of cycle (case sensitive match)"
+          use :bike_attrs
+          optional :components, type: Array do
+            use :components_attrs
+          end
+        end
+        post '/', serializer: BikeV2ShowSerializer, root: 'bike' do
+          declared_p = { "declared_params" => declared(params, include_missing: false).merge(creation_state_params) }
+          b_param = BParam.create(creator_id: creation_user_id, params: {'bike': declared_p['declared_params']}, origin: 'api_v3')
```

Comment:
I'm running into issues here -- a lot of this code is replicated from V2, but for some reason a lot of the code expects `declared_params` to be under the `bike` key, but for some reason neither requests fired through the Swagger documentation nor Rspec coerced it properly. I added this key manually to resolve some errors and begin testing logic in the happy case.